File: 자프링-GraphQL-실습/src/main/java/com/junyss/graphqltest/api/team/model/dto/response/TeamAndMemberResponseDto.java
package com.junyss.graphqltest.api.team.model.dto.response;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import com.junyss.graphqltest.api.people.model.dto.response.PeopleResponseDto;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class TeamAndMemberResponseDto {
	private Long teamId;
	private String manager;
	private String office;
	private String extensionNumber;
	private String mascot;
	private String cleaningDuty;
	private String project;
	private List<PeopleResponseDto> members;

	@Builder
	public TeamAndMemberResponseDto(Long teamId, String manager, String office, String extensionNumber, String mascot,
		String cleaningDuty, String project, List<PeopleResponseDto> peopleResponseDtos) {
		this.teamId = teamId;
		this.manager = manager;
		this.office = office;
		this.extensionNumber = extensionNumber;
		this.mascot = mascot;
		this.cleaningDuty = cleaningDuty;
		this.project = project;
		this.members = Objects.requireNonNullElseGet(peopleResponseDtos, ArrayList::new);
	}
}

File: type-script-nest-graph-ql-실습/src/app/team/model/dto/response/team-and-member-response.dto.ts
import { PeopleResponseDto } from "../../../../people/model/dto/response/people.response.dto";
import { TeamEntity } from "../../entities/team.entity";

export class TeamAndMemberResponseDto {
  private teamId: number;
  private manager: string;
  private office: string;
  private extensionNumber: string;
  private mascot: string;
  private cleaningDuty: string;
  private project: string;
  private members: PeopleResponseDto[];

  constructor(entity: TeamEntity, peopleResponseDto: PeopleResponseDto[]) {
    this.teamId = entity.teamId;
    this.manager = entity.manager;
    this.office = entity.office;
    this.extensionNumber = entity.extensionNumber;
    this.mascot = entity.mascot;
    this.cleaningDuty = entity.cleaningDuty;
    this.project = entity.project;
    this.members = peopleResponseDto || [];
  }
}
